lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1190,14 +1190,16 @@ def _relation_hosts(self, relation: Relation) -> Dict[str, Tuple[str, str]]:
         hosts = {}
         for unit in relation.units:
             # TODO deprecate and remove unit.name
-            unit_name = relation.data[unit].get("prometheus_scrape_unit_name") or unit.name
-            # TODO deprecate and remove "prometheus_scrape_host"
-            unit_address = relation.data[unit].get(
-                "prometheus_scrape_unit_address"
-            ) or relation.data[unit].get("prometheus_scrape_host")
-            unit_path = relation.data[unit].get("prometheus_scrape_unit_path", "")
-            if unit_name and unit_address:
-                hosts.update({unit_name: (unit_address, unit_path)})
+            unit_databag = relation.data.get(unit)
+            if unit_databag:
+                unit_name = unit_databag.get("prometheus_scrape_unit_name") or unit.name
+                # TODO deprecate and remove "prometheus_scrape_host"
+                unit_address = unit_databag.get(
+                    "prometheus_scrape_unit_address"
+                ) or unit_databag.get("prometheus_scrape_host")
+                unit_path = unit_databag.get("prometheus_scrape_unit_path", "")
+                if unit_name and unit_address:
+                    hosts.update({unit_name: (unit_address, unit_path)})
```

Comment:
The most important part of this method is:

`hosts.update({unit_name: (unit_address, unit_path)})`

and it is 3 indentation levels deep.

How about reducing the cognitive load of understanding what this method does with this refactor? 🤔 

```suggestion
            unit_databag = relation.data.get(unit)
            if not unit_databag:
                continue

            unit_name = unit_databag.get("prometheus_scrape_unit_name") or unit.name
            # TODO deprecate and remove "prometheus_scrape_host"
            unit_address = unit_databag.get(
                "prometheus_scrape_unit_address"
            ) or unit_databag.get("prometheus_scrape_host")
            unit_path = unit_databag.get("prometheus_scrape_unit_path", "")

            if not (unit_name and unit_address):
                continue
    
            hosts.update({unit_name: (unit_address, unit_path)})
```